### PR TITLE
[ticket/15300] Show maximum attachment filesize

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-15300

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15300

### CODE REVIEW

**Comment by rubencm** [comment]
Another version of #5173

**Comment by hanakin** [comment]
@rubencm perhaps just rebase on 3.3.x and we can merge this

**Comment by rubencm** [comment]
@hanakin done

**Comment by 3D-I** [comment]
why just 3.3?

File: phpBB/includes/functions_posting.php [hunk 1]
```diff
@@ -726,10 +726,11 @@ function posting_gen_attachment_entry($attachment_data, &$filename_data, $show_a
...
 	$template->assign_vars(array(
-		'S_SHOW_ATTACH_BOX'	=> $show_attach_box,
-		'S_HAS_ATTACHMENTS'	=> count($attachment_data),
-		'FILESIZE'			=> $config['max_filesize'],
-		'FILE_COMMENT'		=> (isset($filename_data['filecomment'])) ? $filename_data['filecomment'] : '',
+		'S_SHOW_ATTACH_BOX'				=> $show_attach_box,
+		'S_HAS_ATTACHMENTS'				=> count($attachment_data),
+		'FILESIZE'						=> $config['max_filesize'],
+		'FILE_COMMENT'					=> (isset($filename_data['filecomment'])) ? $filename_data['filecomment'] : '',
+		'MAX_ATTACHMENT_FILESIZE'		=> $user->lang('MAX_ATTACHMENT_FILESIZE', get_formatted_filesize($config['max_filesize'])),
```
Comment (3D-I) [inline phpBB/includes/functions_posting.php:733]: This will show `0 Bytes` instead of `Unlimited`, in case,  if I am not wrong?

**Comment by hanakin** [comment]
its an improvement which technically should only go into master by definition, but its very minimal so I think we can do it in a 3.3.x update, but 3.2.x is frozen to bug/security fixes only which this is not.

File: phpBB/includes/functions_posting.php (hunk 1 above)
Comment (marc1706) [reply phpBB/includes/functions_posting.php]: Adjusted to not display the info if the limit is set to 0.

**Comment by 3D-I** [comment]
@marc1706 
this deserves more attention, I have reread the ticket and reread this PR. Everything is fine but not great, this does not satisfy the request, it is a panacea. It certainly does not finalize the ticket in its entirety, which you can get IMHO.

File: phpBB/includes/functions_posting.php (hunk 1 above)
Comment (3D-I) [reply phpBB/includes/functions_posting.php]: I do not tend to agree with it though. Sorry.

**Comment by marc1706** [comment]
What part of the request does this not satisfy and what do you not agree with in the current design? Please be more specific, thanks.

**Comment by 3D-I** [comment]
@marc1706 

I'm sorry I didn't answer you right away but I don't know how I missed this notification, the last few weeks have been quite difficult here. I'll come back to the subject as soon as possible or I'll see if I can create a new PR with my suggestions, in the end it's nothing serious and we can move on if you like.

Thanks.

**Pull request merged**